packages/core/src/TagsInput/TagsInput.js

```diff
@@ -83,6 +83,7 @@ const HvTagsInput = (props) => {
   const [validationMessage, setValidationMessage] = useControlled(statusMessage, "");
 
   const [stateValid, setStateValid] = useState(true);
+  const [resetInput, setResetInput] = useState(false);
```

Comment:
I suggest you turn `resetInput` into a ref. We want to perform the "reset" when the value changes, the reset flag is auxiliary. Being a stateful var, the useEffect will also unnecessarily trigger when it returns to false.